drivers/regulator/regulator_common.c

```diff
@@ -183,12 +183,16 @@ int regulator_disable(const struct device *dev)
 	(void)k_mutex_lock(&data->lock, K_FOREVER);
 #endif
 
-	data->refcnt--;
+	if (data->refcnt < 1) {
```

Comment:
I suggest putting this check at line 182 and returning immediately, before the mutex.

Reply:
It decreases nesting, but makes possible a race condition:

1. refcnt is 1
2. Thread 1 and Thread 2 call regulator_disable()
3. Both threads pass "if (data->refcnt < 1)" check since refcnt is 1
4. Both threads decrease refcnt and it's now -1